utils/backoff.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package utils
+
+import (
+	"time"
+)
+
+const (
+	backoffBase uint64 = 128
+	maxAttempts        = 4
+)
+
+// BackoffOperation is executed by Retry.
+// The BackoffOperation will be retried the provided
+// number attempts if returning an error.
+type BackoffOperation func() error
```

Comment:
nit: suggest `BackoffFunc` or `RetryFunc` or perhaps even just `TryFunc`

Reply:
Another thought: eliminate the type definition altogether, and just accept `func() error`? Seems self sufficient.

Reply:
I think dropping the type definition is a great suggestion, because I agree on what @lieut-data is saying that `func() error` is sufficient.